torch/distributed/fsdp/symbolic_trace.py

```diff
@@ -0,0 +1,202 @@
+import functools
```

Comment:
If this is planned as a prototype feature in the coming release, this filename needs to start with an underscore? Please ignore if this is supposed to be public

Reply:
Done!